src/commands/run.rs

```diff
@@ -658,6 +646,16 @@ impl RunCommand {
     }
 
     fn load_module(&self, engine: &Engine, path: &Path) -> Result<Module> {
+        // Do not accept wasmtime subcommand names as the module name
+        let path = match path.to_str() {
+            Some("help") | Some("config") | Some("run") | Some("wast") | Some("compile") => {
+                bail!("module name cannot be the same as a subcommand")
+            }
+            #[cfg(unix)]
+            Some("-") => "/dev/stdin".as_ref(),
+            _ => path,
+        };
```

Comment:
Would it be possible to move these checks for "help" etc. higher up so that they only happen if the argument comes before any "--"? So if someone really wants to run a module named "help" or whatever they could do "wasmtime -- help"?

Reply:
Good point! Clap is the one handling `--` so it's sort of hidden from us, but I think I figured out the right clap configuration to make this work. Now `wasmtime run run` will work, `wasmtime run` won't work, but `wasmtime -- run` will work.